src/server/scripts/EasternKingdoms/zone_elwynn_forest.cpp

```diff
@@ -414,11 +415,503 @@ class spell_quest_extincteur : public SpellScriptLoader
+            if (Player* player = killer->GetCharmerOrOwnerPlayerOrPlayerItself())
+                player->RewardPlayerAndGroupAtEvent(NPC_HOGGER, me);
+        }
+
+        void SummonGeneralHammondClay()
+        {
+            TempSummon* hammond = me->SummonCreature(NPC_GENERAL_HAMMOND_CLAY, generalHammondClayPositions[0]);
+            if (hammond)
+            {
+                _generalHammondGUID = hammond->GetGUID();
+                hammond->CastSpell(hammond, SPELL_TELEPORT_VISUAL_ONLY_1, true);
+                hammond->Mount(DISPLAYID_GENERAL_HAMMOND_CLAYS_MOUNT);
+                hammond->AI()->Talk(YELL_OPENING);
+            }
+            // summon ragamuffins and do text
+            TempSummon* ragamuffin1 = me->SummonCreature(NPC_RAGAMUFFIN, ragamuffinPositions[0], TEMPSUMMON_TIMED_DESPAWN, 5000);
+            if (ragamuffin1)
+            {
+                ragamuffin1->AI()->Talk(SAY_CLAY);
+            }
+            TempSummon* ragamuffin2 = me->SummonCreature(NPC_RAGAMUFFIN, ragamuffinPositions[1], TEMPSUMMON_TIMED_DESPAWN, 5000);
+            if (ragamuffin2)
+            {
+                ragamuffin2->AI()->Talk(SAY_WOW);
+            }
+        }
+
+        void MoveGeneralHammondClay()
+        {
+            if (GetHammond())
+            {
+                GetHammond()->SetWalk(true);
+                GetHammond()->GetMotionMaster()->MovePoint(0, generalHammondClayPositions[2], true);
+                _events.ScheduleEvent(EVENT_DISMOUNT_HAMMOND_CLAY, 8500);
+            }
+        }
+
+        void SummonAndromath()
+        {
+            TempSummon* andromath = me->SummonCreature(NPC_ANDROMATH, andromathPositions[0]);
+            if (andromath)
+            {
+                _andromathGUID = andromath->GetGUID();
+                andromath->CastSpell(andromath, SPELL_TELEPORT_VISUAL_ONLY_1, true);
+            }
+        }
+
+        void MoveAndromath()
+        {
+            if (GetAndromath())
+            {
+                GetAndromath()->SetWalk(true);
+                GetAndromath()->GetMotionMaster()->MovePoint(0, andromathPositions[2], true);
+            }
+        }
+
+        void SummonDumas()
+        {
+            TempSummon* dumas = me->SummonCreature(NPC_DUMAS, dumasPositions[0]);
+            if (dumas)
+            {
+                _dumasGUID = dumas->GetGUID();
+                dumas->CastSpell(dumas, SPELL_TELEPORT_VISUAL_ONLY_1, true);
+            }
+        }
+
+        void MoveDumas()
+        {
+            if (GetDumas())
+            {
+                GetDumas()->SetWalk(true);
+                GetDumas()->GetMotionMaster()->MovePoint(0, dumasPositions[2], true);
+            }
+        }
+
+        void SummonMinions()
+        {
+            me->CastStop();
+            Talk(SAY_HOGGER_SUMMON_MINIONS);
+            //DoCastSelf(SPELL_SUMMON_MINIONS, true); This works, but the minions just sit there, and then despawn
+            Position hogPos1 = me->GetPosition();
```

Comment:
If i understand correctly the only thing changing here is the spawn distance, you should use a loop to avoid code repetition

for (float distance : { 0.5f, 1.5f, 2.5f })
{
    // Code here
}